helix-term/src/commands/lsp.rs

```diff
@@ -147,12 +202,20 @@ impl ui::menu::Item for PickerDiagnostic {
             }
         };
 
-        Spans::from(vec![
-            Span::raw(path),
-            Span::styled(&self.diag.message, style),
-            Span::styled(code, style),
-        ])
-        .into()
+        if let Some(icon) = icon {
+            Row::new(vec![
+                icon.into(),
+                Span::raw(path),
+                Span::styled(&self.diag.message, style),
+                Span::styled(code, style),
+            ])
+        } else {
+            Row::new(vec![
+                Span::raw(path),
+                Span::styled(&self.diag.message, style),
+                Span::styled(code, style),
+            ])
+        }
```

Comment:
Untested, but I think the if else block can be replaced by

```rust
        Row::new(vec![
            icon.map(Span::from).unwrap_or_else(|| Span::raw("")),
            Span::raw(path),
            Span::styled(&self.diag.message, style),
            Span::styled(code, style),
        ]);
```

provided that an empty span doesn't take up any space.

Reply:
I just tested this, unfortunately an empty `Span` does take up space. I think it's fine to leave it as is.